src/main/ruby/core/truffle/cext.rb

```diff
@@ -127,6 +127,15 @@ def data_holder
   class RbEncoding
     NAME_FIELD_INDEX = 0
 
+    @cache = {}
+    @mutex = Mutex.new
```

Comment:
These could be constants, it would avoid a couple extra reads and use simpler nodes.
Probably doesn't matter here, though.
Style-wise it would indicate they are no re-assigned.